### PR TITLE
Feat/error boundry

### PR DESCRIPTION
**Pull request by sumitnitsurat**
- error boundary added for template rendering

### CODE REVIEW

**Comment by rjchow** [comment]
spellcheck, boundary

**Comment by rjchow** [comment]
implementation looks good to me but I think visually it could use a bit more work - not really sure what we can do about that.

The visual transition is very jarring, and the font looks off

File: .eslintrc [hunk 1]
```diff
@@ -42,5 +43,13 @@
     "chai-expect/missing-assertion": "error",
     "chai-expect/terminating-properties": "error",
     "prefer-destructuring": "off"
-  }
+    },
+    "overrides": [
```
Comment (rjchow) [inline .eslintrc]: i think its better to override in the file itself so that you can write a comment why it's needed

**Comment by rjchow** [comment]
any progress on this?

File: .eslintrc (hunk 1 above)
Comment (sumitnitsurat) [reply .eslintrc]: To test the buggy component I had to throw the error from the child component. Which causing the console.error to fire to print the error on console. So I override console.error to empty in my test. But for eslint to stop throwing error I had to override here.

**Comment by sumitnitsurat** [comment]
I will ask @samuellye98  to fix the design and colors.

File: src/components/ErrorBoundary.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    });
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <div className={`${css.box}`}>
+          <div id="error">
+            <div className={`${css.error}`}>
+              {/* <i className={`fas fa-exclamation-triangle ${css.icon}`} /> */}
```
Comment (sumitnitsurat) [inline src/components/ErrorBoundary.js]: remove comments if not needed

File: src/components/error.scss [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+.box {
+  width: 100%;
+  height: 100%;
+  min-width: 100%;
+  min-height: calc(100vh - 180px);
+  position: relative;
+  display: inline-block;
+  padding-top: 150px;
+  padding-bottom: 150px;
```
Comment (sumitnitsurat) [inline src/components/error.scss]: this padding is causing the footer to hide at the bottom. Prefer to not have scroll bar and fit the error message inside the page.

File: src/components/error.scss [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  margin-left: auto;
+  margin-right: auto;
+  max-width: 1200px;
+  width: 100%;
+  text-align: center;
+  height: auto;
+  line-height: 1.4;
+}
+
+.error h2 {
```
Comment (sumitnitsurat) [inline src/components/error.scss]: use scss styling guidelines. nested stling.

File: src/components/ErrorBoundary.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+              <img
+                src="/static/images/errorpage/error.png"
+                style={{ height: "15vh" }}
+              />
+              <h2>Something went wrong!</h2>
+              <p>
+                There is an error with this certificate, please contact your
+                issuing institution.
+              </p>
+              <a href="/index">Go Back</a>
```
Comment (sumitnitsurat) [inline src/components/ErrorBoundary.js]: just put "/" to redirect. dont need to specify the index.

File: src/components/Layout/FooterBar.js [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
-      <div>
-        <a href="/registry">Registry</a>
-      </div>
-      <div>
-        <a href="/privacy">Privacy Policy</a>
-      </div>
-      <div>
-        <a href="https://github.com/OpenCerts">Github</a>
-      </div>
+  <div className={css.footer}>
```
Comment (sumitnitsurat) [inline src/components/Layout/FooterBar.js]: footer is distorted and expanded. Check out at opencerts.io.

File: .eslintrc (hunk 1 above)
Comment (Nebulis) [reply .eslintrc]: Oops

Comment (Nebulis) [reply .eslintrc]: You can disable eslint as a one liner

File: src/components/error.scss (hunk 4 above)
Comment (Nebulis) [reply src/components/error.scss]: Or don't use sass nor nested styling 😊

File: .eslintrc (hunk 1 above)
Comment (sumitnitsurat) [reply .eslintrc]: fixed it

File: src/components/errorboundary.test.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { mount } from "enzyme";
+import ErrorBoundary from "./ErrorBoundary";
+
+const ProblemChild = () => {
+  throw new Error("Error thrown from problem child");
+};
+/* eslint-disable no-console */
```
Comment (rjchow) [inline src/components/errorboundary.test.js]: good practice to leave a comment as to why eslint was disabled in this instance

Comment (rjchow) [reply src/components/errorboundary.test.js]: good practice to leave a comment as to why eslint was disabled in this instance

File: src/components/errorboundary.test.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { mount } from "enzyme";
```
Comment (rjchow) [inline src/components/errorboundary.test.js]: test filename doesn't match filename of component under test - capitalisation

**Comment by rjchow** [comment]
only two minor issues can merge after resolving it

File: src/components/errorboundary.test.js (hunk 7 above)
Comment (samuellye98) [reply src/components/errorboundary.test.js]: disabling lint for the overridden console

**Pull request merged**